features/search-replace.feature

```diff
@@ -1030,3 +1030,24 @@ Feature: Do global search/replace
       """
       Success: 1 replacement to be made.
       """
+
+  Scenario: Regex search/replace with `--regex-limit=1` option
+    Given a WP install
+    And I run `wp post create --post_content="I have a pen, I have an apple. Pen, pine-apple, apple-pen."`
+
+    When I run `wp search-replace --regex "ap{2}le" "orange" --regex-limit=1 --log`
+    Then STDOUT should contain:
+    """
+    I have a pen, I have an orange. Pen, pine-apple, apple-pen.
+    """
+
+
+  Scenario: Regex search/replace with `--regex-limit=2` option
+    Given a WP install
+    And I run `wp post create --post_content="I have a pen, I have an apple. Pen, pine-apple, apple-pen."`
+
+    When I run `wp search-replace --regex "ap{2}le" "orange" --regex-limit=2 --log`
+    Then STDOUT should contain:
+    """
+    I have a pen, I have an orange. Pen, pine-orange, apple-pen.
+    """
```

Comment:
Could you add a scenario to test for incorrect args here, eg
```
  Scenario: Regex search/replace with incorrect or default `--regex-limit`
    Given a WP install
    When I try `wp search-replace '(Hello)\s(world)' '$2, $1' --regex --regex-limit=asdf`
    Then STDERR should be:
      """
      Error: `--regex-limit` expects a non-zero positive integer or -1.
      """
    When I try `wp search-replace '(Hello)\s(world)' '$2, $1' --regex --regex-limit=0`
    Then STDERR should be:
      """
      Error: `--regex-limit` expects a non-zero positive integer or -1.
      """
    When I try `wp search-replace '(Hello)\s(world)' '$2, $1' --regex --regex-limit=-2`
    Then STDERR should be:
      """
      Error: `--regex-limit` expects a non-zero positive integer or -1.
      """
    When I run `wp search-replace '(Hello)\s(world)' '$2, $1' --regex --regex-limit=-1`
    Then STDOUT should contain:
      """
      Success:
      """
```
(See below for disallowing zero.)